src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -312,6 +261,94 @@ private void ProcessObjectContent(PSObject inputObject)
                     WriteHexidecimal(inputBytes, null, 0);
                 }
             }
+            else
+            {
+                string errorMessage = StringUtil.Format(UtilityCommonStrings.FormatHexTypeNotSupported, obj.GetType());
+                ErrorRecord errorRecord = new ErrorRecord(new ArgumentException(errorMessage),
+                                                            "FormatHexTypeNotSupported",
+                                                            ErrorCategory.InvalidArgument,
+                                                            obj.GetType());
+                WriteError(errorRecord);
+            }
+        }
+
+        private byte[] ConvertToByteArray(object inputObject)
+        {
+            if (inputObject is string str)
+            {
+                return Encoding.GetBytes(str);
+            }
+
+            var baseType = inputObject.GetType();
+            byte[] result = null;
+            int elements = 1;
+            bool isArray = false;
+            bool isBool = false;
+            bool isEnum = false;
+            if (baseType.IsArray)
+            {
+                baseType = baseType.GetElementType();
+                dynamic o = inputObject;
+                elements = (int)o.Length;
+                isArray = true;
+            }
+
+            if (baseType.IsEnum)
+            {
+                baseType = baseType.GetEnumUnderlyingType();
+                isEnum = true;
+            }
+
+            if (baseType.IsPrimitive && elements > 0)
+            {
+                if (baseType == typeof(bool))
+                {
+                    isBool = true;
+                }
+
+                var elementSize = Marshal.SizeOf(baseType);
```

Comment:
I could be wrong, but wasn't one of @lzybkr's comments that this would return the wrong size for boolean values?

Reply:
I think it is ok - only local bool var-s have 4 byte size, rest - 1 byte.